mailpile/plugins/vcard_face.py

```diff
@@ -0,0 +1,57 @@
+import time
+
+from mailpile.i18n import gettext as _
+from mailpile.mailutils import Email
+from mailpile.plugins import PluginManager
+from mailpile.search import MailIndex
+from mailpile.vcard import MailpileVCard, VCardImporter, VCardLine
+
+
+_plugins = PluginManager(builtin=__file__)
+
+
+class FaceImporter(VCardImporter):
+    """
+    Get avatar from Face header.
+    """
+    FORMAT_NAME = 'Face'
+    FORMAT_DESCRIPTION = _('Get avatar from Face header.')
+    SHORT_NAME = 'face'
+    CONFIG_RULES = {
+        'active': [_('Enable this importer'), bool, True],
+    }
+    VCARD_TS = 'x-face-ts'
+    VCARD_IMG = ''
+
+    def get_vcards(self):
+        mail_index = MailIndex(self.session.config)
```

Comment:
This is not the right way to get a mail index object - you've just created an entire new index and loaded from disk. For a user with a lot of mail, this will take a very long time and waste a lot of computing time. Instead:

    mail_index = self.session.config.get_index(self.session)